### PR TITLE
api:Provide a default implementation for new method added to ManagedTranport.Listener to support ClientTransportFilters

### PR DESCRIPTION
**Pull request by larry-safran**
Avoids breaking backward compatibility which was done by #10646.

### CODE REVIEW

File: core/src/main/java/io/grpc/internal/ManagedClientTransport.java [hunk 1]
```diff
@@ -110,6 +110,8 @@ interface Listener {
      * Called just before {@link #transportReady} to allow direct modification of transport
      * Attributes.
      */
-    Attributes filterTransport(Attributes attributes);
+    default Attributes filterTransport(Attributes attributes) {
```
Comment (sergiitk) [inline core/src/main/java/io/grpc/internal/ManagedClientTransport.java:113]: I was kinda on a fence about adding a default implementation to a pure interface, but after talking IRL I see how it makes sense. It's an optional filter, so I think we're all good here.

File: core/src/main/java/io/grpc/internal/ManagedClientTransport.java [hunk 2]
```diff
@@ -110,6 +110,8 @@ interface Listener {
...
+      return attributes;
```
Comment (sergiitk) [inline core/src/main/java/io/grpc/internal/ManagedClientTransport.java:114]: Minor: just for the sake of code coverage numbers, let's add a simple unit test.

File: api/src/test/java/io/grpc/ContextTest.java [hunk 3]
```diff
@@ -706,7 +706,7 @@ public void relativeDeadlineTriggersAndPropagates() throws Exception {
 
   @Test
   public void innerDeadlineCompletesBeforeOuter() throws Exception {
-    Context base = Context.current().withDeadline(Deadline.after(2, TimeUnit.SECONDS), scheduler);
+    Context base = Context.current().withDeadline(Deadline.after(3, TimeUnit.SECONDS), scheduler);
```
Comment (sergiitk) [inline api/src/test/java/io/grpc/ContextTest.java:709]: Spoke IRL: this change is unrelated, but macos test breaks with lower timeouts. I'm good with that.

**Review by sergiitk — APPROVED** [review]

File: core/src/main/java/io/grpc/internal/ManagedClientTransport.java (hunk 2 above)
Comment (larry-safran) [reply core/src/main/java/io/grpc/internal/ManagedClientTransport.java:114]: Done.

**Pull request merged**